### DISCUSSION TITLE
Как можно фильтровать Ошибки в 3х

### DISCUSSION THREAD

**Discussion by zufar-ik** [aiogram/aiogram#1104] [Q&A] [answered]
как фильтровать ошибки в 3х 
Мне надо отфильтровать ошибки как ```CantParseEntities, MessageTextIsEmpty, RetryAfter, ChatNotFound```
В 2х по умолчанию были отльные ошибки а в 3х все ошибки это ```TelegramBadRequest```
Кроме как из текста брать неть норм способов?

**Comment by urlykkis** [comment]
@dp.error_handler()

**Comment by Olegt0rr** [comment, marked as answer]
К сожалению, Telegram Bot API не представляет возможности цивилизованно фильтровать ошибки. По коду статуса большинство ошибок легко спутать, а описание ошибок регулярно меняется без какого-либо предупреждения.

Упомянутую вами ошибку `TelegramRetryAfter` можно ловить на уровне aiogram (т.к. у неё уникальный код статуса)
